acceptance/helpers/delete_clusters/delete_clusters.go

```diff
@@ -0,0 +1,227 @@
+		if err != nil {
+			fmt.Println("Error: ", err, out)
+			exists = false
+		} else {
+			if out == "[]" || out == "[]\n" {
+				fmt.Println("AKS cluster does not exisit: " + aks_resource_group + runID)
+				exists = false
+			} else {
+				exists = true
+			}
+		}
+	case "EKS":
+		eks_region := os.Getenv("EKS_REGION")
+		out, err := proc.RunW("aws", "eks", "list-clusters", "--region="+eks_region, "--query", "clusters | [? contains(@,'epinio-ci"+runID+"')]", "--output", "text")
+		if err != nil {
+			fmt.Println("Error: ", err, out)
+			exists = false
+		} else {
+			if out == "" {
+				fmt.Println("EKS cluster does not exisit: epinio-ci" + runID)
+				exists = false
+			} else {
+				exists = true
+			}
+		}
+	case "GKE":
+		gke_zone := os.Getenv("GKE_ZONE")
+		os.Setenv("USE_GKE_GCLOUD_AUTH_PLUGIN", "true")
+		out, err := proc.RunW("gcloud", "container", "clusters", "list", "--filter", "epinioci"+runID, "--zone", gke_zone, "--quiet")
+		if err != nil {
+			fmt.Println("Error: ", err, out)
+			exists = false
+		} else {
+			if out == "" {
+				fmt.Println("GKE cluster does not exisit: epinioci" + runID)
+				exists = false
+			} else {
+				exists = true
+			}
+		}
+	}
+	return exists
+}
+
+func GetKubeconfig(runID string, pcp string) {
+	kubeconfig_name := os.Getenv("KUBECONFIG_NAME")+"-Deletion"
+
+	switch pcp {
+		case "AKS":
+			aks_resource_group := os.Getenv("AKS_RESOURCE_GROUP")
+			out, err := proc.RunW("az", "aks", "get-credentials", "--admin", "--resource-group", aks_resource_group, "--name", aks_resource_group+runID, "--file", kubeconfig_name)
+			if err != nil {
+				fmt.Println("Error: ", err, out)
+			} else {
+				fmt.Println("Fetched current kubeconfig")
+			}
+		case "EKS":
+			eks_region := os.Getenv("EKS_REGION")
+			out, err := proc.RunW("eksctl", "utils", "write-kubeconfig", "--region", eks_region, "--cluster", "epinio-ci"+runID, "--kubeconfig", kubeconfig_name)
+			if err != nil {
+				fmt.Println("Error: ", err, out)
+			} else {
+				fmt.Println("Fetched current kubeconfig")
+			}
+		case "GKE":
+			gke_zone := os.Getenv("GKE_ZONE")
+			epci_gke_project := os.Getenv("EPCI_GKE_PROJECT")
+			os.Setenv("USE_GKE_GCLOUD_AUTH_PLUGIN", "true")
+			os.Setenv("KUBECONFIG", kubeconfig_name)
+			out, err := proc.RunW("gcloud", "container", "clusters", "get-credentials", "epinioci"+runID, "--zone", gke_zone, "--project", epci_gke_project)
+			if err != nil {
+				fmt.Println("Error: ", err, out)
+			} else {
+				fmt.Println("Fetched current kubeconfig")
+			}
+		}
+}
+
+func CleanupNamespaces() {
+	kubeconfig_name := os.Getenv("KUBECONFIG_NAME")+"-Deletion"
+	os.Setenv("KUBECONFIG", kubeconfig_name)
+
+	out, err := proc.RunW("kubectl", "--kubeconfig", kubeconfig_name, "delete", "--force", "--ignore-not-found", "namespace", "epinio", "workspace")
+	if err != nil {
+		fmt.Println("Error: ", err, out)
+	} else {
+		fmt.Println("Cleaning up test namespaces ...")
+	}
+}
+
+func DeleteCluster(runID string, pcp string) {
```

Comment:
I've seen that the `pcp` is used along all the other func always to switch over the different implementations.
This is a bit repeating, and harder to read. I would just split once the logic, and have different functions:

```go
func DeleteCluster(runID string, pcp string) {
//
	switch pcp {
	case "AKS":
		deleteAKSCluster(runID)
	case "EKS":
		deleteEKSCluster(runID)
	case "GKE":
		deleteGKECluster(runID)
	}
}
```

Reply:
Addressed.